### DISCUSSION TITLE
Memory channel doesn't clear queues on close

### DISCUSSION THREAD

**Discussion by karolinepauls** [celery/kombu#1474] [Q&A]
https://github.com/celery/kombu/blob/cbd327dc3bdd124c1bc4fc3d8f2b451fd10e2da6/kombu/transport/memory.py#L59

`queue.empty()` doesn't empty the queue - it returns a boolean stating whether the queue is empty. In order to clear a queue (without touching the internals), one has to pop from it until it's empty.

It remains a question if clearing should take place at all and whether the queues should be channel-scoped or transport-scoped. There's definitely a need to be able to clear the queues between tests.

**Comment by auvipy** [comment]
failing test case with a fix?

**Comment by karolinepauls** [comment]
@auvipy There is no fix because I don't know what the expected behaviour is.

> It remains a question if clearing should take place at all and whether the queues should be channel-scoped or transport-scoped. There's definitely a need to be able to clear the queues between tests.

It is crucial to decide on the behaviour wanted before commencing implementation.

**Comment by thedrow** [comment]
This is definitely a mistake.
Wouldn't setting `self.queues` to an empty dict be enough?

**Comment by karolinepauls** [comment]
The question remains - channel-scoped or transport-scoped? If it's answered, the fix is trivial.

**Comment by thedrow** [comment]
If you purge the queue using RabbitMQ, it is purged for all channels, right?
